core/templates/components/skill-mastery/skill-mastery.component.ts

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the skill mastery viewer.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { SkillMasteryListConstants } from
+  'components/skills-mastery-list/skills-mastery-list.constants';
+import { SkillMasteryBackendApiService } from
+  'domain/skill/skill-mastery-backend-api.service';
+
+@Component({
+  selector: 'skill-mastery-viewer',
+  templateUrl: './skill-mastery.component.html',
+  styleUrls: []
+})
+export class SkillMasteryViewerComponent implements OnInit {
+  @Input() skillId: string;
+  @Input() masteryChange: number;
+
+  skillMasteryDegree: number = 0;
+
+  constructor(
+    private skillMasteryBackendApiService: SkillMasteryBackendApiService
+  ) {}
+
+  ngOnInit() {
+    this.skillMasteryDegree = 0.0;
+
+    this.skillMasteryBackendApiService.fetchSkillMasteryDegrees(
+      [this.skillId]).then(degreesOfMastery => this.skillMasteryDegree = (
+      degreesOfMastery[this.skillId]));
+  }
+
+  getSkillMasteryPercentage() {
+    return Math.round(this.skillMasteryDegree * 100);
+  }
+
+  getMasteryChangePercentage() {
+    if (this.masteryChange >= 0) {
+      return '+' + Math.round(this.masteryChange * 100);
+    } else {
+      return Math.round(this.masteryChange * 100);
+    }
+  }
+
+  getLearningTips() {
+    if (this.masteryChange > 0) {
+      if (this.skillMasteryDegree >=
+          SkillMasteryListConstants.MASTERY_CUTOFF.GOOD_CUTOFF) {
+        return ('You have mastered this skill very well! ' +
+          'You can work on other skills or learn new skills.');
+      } else {
+        return ('You have made progress! You can increase your ' +
+          'mastery level by doing more practice sessions.');
+      }
+    } else {
+      return ('Looks like your mastery of this skill has dropped. ' +
+          'To improve it, try reviewing the concept card below and ' +
+          'then practicing more questions for the skill.');
+    }
+  }
```

Comment:
I know you didn't write this code (not blaming you). Doesn't the suggested code look cleaner? The indentation of my suggested code might be off but take a look once. Use guard clauses, they will make your code a lot cleaner and easy to read. In case you don't know about guard clauses, you can learn more about them here: https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html 
```suggestion
    if (this.this.masteryChange <= 0) {
      return ('Looks like your mastery of this skill has dropped. ' +
      'To improve it, try reviewing the concept card below and ' +
      'then practicing more questions for the skill.');
    }
    if (this.skillMasteryDegree >=
      SkillMasteryListConstants.MASTERY_CUTOFF.GOOD_CUTOFF) {
      return ('You have mastered this skill very well! ' +
      'You can work on other skills or learn new skills.');
    }
    return ('You have made progress! You can increase your ' +
            'mastery level by doing more practice sessions.');
  }
```

Reply:
Yup, looks better. Done.